data_pipeline/gmail_data/get_gmail_data.py

```diff
@@ -0,0 +1,45 @@
+from typing import Iterable
+import pandas as pd
+
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.discovery import Resource
+
+
+# this connection type will be changed !!
+def connect_to_email(userId: str) -> Resource:
+    SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
```

Comment:
(may have commented before but can't see it)
I would suggest to move all constants out of the function (may just be this one). In this case maybe also add a prefix, e.g. `GMAIL_SCOPES`